contracts/eth/nearbridge/contracts/NearBridge.sol

```diff
@@ -12,57 +13,39 @@ contract NearBridge is INearBridge, AdminControlled {
     using Borsh for Borsh.Data;
     using NearDecoder for Borsh.Data;
 
-    struct BlockProducer {
-        NearDecoder.PublicKey publicKey;
-        uint128 stake;
-    }
-
-    // Information about the block producers of a certain epoch.
-    struct BlockProducerInfo {
-        uint256 bpsLength;
-        uint256 totalStake;
-        mapping(uint256 => BlockProducer) bps;
-    }
+    // Assumed to be even and to not exceed 256.
+    uint constant MAX_BLOCK_PRODUCERS = 100;
 
-    // Minimal information about the submitted block.
-    struct BlockInfo {
-        uint64 height;
-        uint256 timestamp;
+    struct Epoch {
         bytes32 epochId;
-        bytes32 nextEpochId;
-        bytes32 hash;
-        bytes32 merkleRoot;
-        bytes32 next_hash;
+        uint numBPs;
+        bytes32[MAX_BLOCK_PRODUCERS] keys;
+        bytes32[MAX_BLOCK_PRODUCERS / 2] packedStakes;
+        uint256 stakeThreshold;
     }
 
     // Whether the contract was initialized.
     bool public initialized;
-    // The `0` address where we are going to send half of the bond when challenge is successful.
-    address payable burner;
     uint256 public lockEthAmount;
     uint256 public lockDuration;
     // replaceDuration is in nanoseconds, because it is a difference between NEAR timestamps.
     uint256 public replaceDuration;
-    Ed25519 edwards;
-
-    // Block producers of the current epoch.
-    BlockProducerInfo currentBlockProducers;
-    // Block producers of the next epoch.
-    BlockProducerInfo nextBlockProducers;
+    Ed25519 immutable edwards;
 
-    // The most recent head that is guaranteed to be valid.
-    BlockInfo head;
+    Epoch[3] epochs;
```

Comment:
Why using size `3` instead of `2`? 

In case there is a good reason I'm missing, what about using size `4`, so we avoid using `% 3` through the code, and instead use `& 3` (which is the same as `% 4).